client/dashboard/profile-wizard/steps/business-details.js

```diff
@@ -299,7 +299,7 @@ class BusinessDetails extends Component {
 									isInstallingExtensions: false,
 								} );
 							} }
-							autoInstall
+							autoInstall={ ! extensionInstallError }
```

Comment:
I think this kind of logic could be moved inside the component, but I think if we do this just to avoid returning `null` in the render, we can move up the newly added `Continue` button in the `render()` method of `Plugins` to avoid doing this.

Reply:
Great suggestion, and it does work. Just to note that it is possible for the button to disappear doing it this way by choosing the another plugin to install:

![Screen Shot 2020-02-28 at 4 42 45 PM](https://user-images.githubusercontent.com/1922453/75508717-ef2c6d00-5a49-11ea-9f4d-2e1bba965503.png)

In the screen shot above, Mailchimp failed and then I switched the toggles such that FB is now selected. `hasErrors` is false and the length of plugins is greater than one. Granted, this is edge case. What do you think?

Reply:
Great catch, Paul!  I think it may be an edge case, but still worth addressing.

After diving into this a bit more, I think the real issue here is that this `Plugin` component should not have `autoInstall` since it does not in fact auto-install.  This will require a few changes, but actually simplifies a lot of the logic in this component:

* Remove `autoInstall` prop.
* Add `this.onContinue( values );` to `onComplete` and `onSkip` functions.
* Remove all references to `extensionInstallError` and `isInstallingExtensions` in state.
* Make sure `goToNextStep();` is hit in `onContinue` if no error exists (removing `0 === businessExtensions.length` logic.
* Add continue text and skip text props to `Plugin` component.

This results in a lot more red and tests well on my end.  Feel free to ping me if you have any questions.

Reply:
https://github.com/woocommerce/woocommerce-admin/pull/3787/commits/5b9386957a501ac3e729e5f16b0ee71c9d792859 Applies your suggestions. Let me know if I'm missing something, but something is off when I try to install a plugin without forcing a server error.

If I click "Continue", the async `onContinue` function is called, but it gets executed synchronously, which I thought was odd. Sure enough, on `version/1.0` branch this function performs this line synchronously, `await updateProfileItems( updates );`.

https://github.com/woocommerce/woocommerce-admin/blob/5b9386957a501ac3e729e5f16b0ee71c9d792859/client/dashboard/profile-wizard/steps/business-details.js#L93

It seems the old logic, that you suggested I remove is what keeps this function appearing "async" to the user:

>Make sure goToNextStep(); is hit in onContinue if no error exists (removing 0 === businessExtensions.length logic.

`updateProfileItems` is a Fresh Data action that is not async, so `await` does nothing in this case. FWIW, the [settings data store refactor](https://github.com/woocommerce/woocommerce-admin/pull/3313 ) exports actions that are defined as aync.

Am I looking at this and implemented your suggestion correctly here?

Reply:
>updateProfileItems is a Fresh Data action that is not async, so await does nothing in this case. FWIW, the settings data store refactor exports actions that are defined as aync.

Unfortunately we have a bunch of these that are incorrectly set up (see https://github.com/woocommerce/woocommerce-admin/issues/2959).  Async functions will make this process immensely easier.


>If I click "Continue", the async onContinue function is called, but it gets executed synchronously, which I thought was odd. Sure enough, on version/1.0 branch this function performs this line synchronously:

Hmm, I am seeing an issue, but I'm seeing it performed async with these changes which goes to the next step before the plugins have a chance to be fully installed and activated.  

The `goToNextStep()`should probably be called after we receive both an `onComplete` from the plugins and profile items have finished updating inside `componentDidUpdate()`.  I think this is a worthwhile refactor to this step, but if you want to use the previous `autoInstall` option, I think that's an okay temporary solution if we want to open an issue to update this later.

Reply:
>Unfortunately we have a bunch of these that are incorrectly set up (see #2959). Async functions will make this process immensely easier.

Oh yeah, I see that now.

>Hmm, I am seeing an issue, but I'm seeing it performed async with these changes which goes to the next step before the plugins have a chance to be fully installed and activated.

Cool, I think we are on the same page on this one. I confuse myself with sync/async.

>The goToNextStep()should probably be called after we receive both an onComplete from the plugins and profile items have finished updating inside componentDidUpdate(). I think this is a worthwhile refactor to this step, but if you want to use the previous autoInstall option, I think that's an okay temporary solution if we want to open an issue to update this later.

I agree that this is a worthwhile refactor. The problem is that `onComplete` is called when the user clicks "Continue", so `BusinessDetails` would need a `componentDidUpdate` if statement to check if profileItems are done requesting (this happens twice), if plugin instillation is done requesting (would need to create a prop on `Plugins` a la `onSuccess`), and possibly if plugin activation is done requesting before calling `goToNextStep`.

With our data store refactors on the horizon, my preference is to keep the `autoInstall` check in this PR because true async functions will clean up the logic so much and we will be refactoring this soon anyways.

if that sounds good to you, I'll revert that last commit.

Reply:
> if that sounds good to you, I'll revert that last commit.

Ideally I'd like this to get wrapped up and landed in `version/1.0` since it is a reproducible bug that folks are likely hitting in production right now.

Reply:
>Ideally I'd like this to get wrapped up and landed in version/1.0 since it is a reproducible bug that folks are likely hitting in production right now.

This is back to its original state that @joshuatf tested. I'll leave it for one last look, but feel free to merge if I'm not online.